util/kustomize/kustomize.go

```diff
@@ -135,6 +143,26 @@ func (k *kustomize) Build(opts *v1alpha1.ApplicationSourceKustomize) ([]*unstruc
 		return nil, nil, nil, err
 	}
 	defer func() { _ = closer.Close() }()
+
+	// If we were passed a HTTPS URL, make sure that we also check whether there
+	// is a custom CA bundle configured for connecting to the server.
+	if k.repo != "" && git.IsHTTPSURL(k.repo) {
+		parsedURL, err := url.Parse(k.repo)
+		if err != nil {
+			log.Warnf("Could not parse URL %s: %v", k.repo, err)
+		} else {
+			caPath, err := certutil.GetCertBundlePathForRepository(parsedURL.Host)
+			if err != nil {
+				log.Warnf("Could not get CA bundle path for %s: %v", parsedURL.Host, err)
+			} else if caPath == "" {
+				log.Debugf("No caCert found for repo %s", parsedURL.Host)
+			} else {
+				log.Debugf("Using Kustomize CA Path: %s", caPath)
```

Comment:
not Kustomize? Git?

Reply:
I'm not sure if I get what you mean by that.

Reply:
the `caPath` is used by Git, not by Kusotimze (thought it is Kustomize that invokes it)

Reply:
Clarified & cleaned up.